app/code/Magento/InventoryReservationCli/Command/ShowInconsistencies.php

```diff
@@ -146,33 +174,78 @@ private function rawOutput(OutputInterface $output, array $inconsistencies): voi
     }
 
     /**
-     * {@inheritdoc}
+     * @inheritdoc
      *
      * @param InputInterface $input
      * @param OutputInterface $output
      * @return int
+     * @throws LocalizedException
+     * @throws ValidationException
+     * @throws SkuIsNotAssignedToStockException
      */
     public function execute(InputInterface $input, OutputInterface $output): int
     {
-        $inconsistencies = $this->getSalableQuantityInconsistencies->execute();
+        $startTime = microtime(true);
+        $isRawOutput = (bool)$input->getOption('raw');
+        $bunchSize = (int)$input->getOption('bunch-size');
 
-        if ($input->getOption('complete-orders')) {
-            $inconsistencies = $this->filterCompleteOrders->execute($inconsistencies);
-        } elseif ($input->getOption('incomplete-orders')) {
-            $inconsistencies = $this->filterIncompleteOrders->execute($inconsistencies);
+        $maxPage = $this->retrieveMaxPage($bunchSize);
+        $hasInconsistencies = false;
+
+        for ($page = 1; $page <= $maxPage; $page++) {
+            $startBunchExecution = microtime(true);
+
+            $inconsistencies = $this->getSalableQuantityInconsistencies->execute($bunchSize, $page);
+            if ($input->getOption('complete-orders')) {
+                $inconsistencies = $this->filterCompleteOrders->execute($inconsistencies);
+            } elseif ($input->getOption('incomplete-orders')) {
+                $inconsistencies = $this->filterIncompleteOrders->execute($inconsistencies);
+            }
+
+            $hasInconsistencies = true;
```

Comment:
Current logic looks like broken. `$hasInconsistencies` will be also true in case when `$inconsistencies` is empty.

Reply:
changed